tests/core/string/test_string.h

```diff
@@ -553,6 +553,53 @@ TEST_CASE("[String] String to integer") {
 	ERR_PRINT_ON
 }
 
+TEST_CASE("[String] String to unsigned integer") {
+	// Test normal valid cases
+	CHECK(String("1237461283").to_uint64() == 1237461283);
+	CHECK(String("0").to_uint64() == 0);
+	CHECK(String("").to_uint64() == 0);
+	CHECK(String("10_000_000").to_uint64() == 10);
+	CHECK(String("10__000").to_uint64() == 10);
+	CHECK(String("  1  2  34 ").to_uint64() == 1);
+	CHECK(String("007").to_uint64() == 7);
+	CHECK(String("18446744073709551615").to_uint64() == 18446744073709551615ULL);
+	CHECK(String("+123").to_uint64() == 123);
+	CHECK(String(" + 123 ").to_uint64() == 123);
+
+	// Test negative numbers and out of range values
+	ERR_PRINT_OFF;
+	CHECK(String("-1").to_uint64() == 0);
+	CHECK(String("-9223372036854775808").to_uint64() == 0);
+	CHECK(String("-9223372036854775809").to_uint64() == 0);
+	CHECK(String("-18446744073709551615").to_uint64() == 0);
+	CHECK(String("-18446744073709551616").to_uint64() == 0);
+	CHECK(String("18446744073709551616").to_uint64() == UINT64_MAX);
+	CHECK(String("18446744073709551620").to_uint64() == UINT64_MAX);
+	CHECK(String("999999999999999999999999999999999999").to_uint64() == UINT64_MAX);
+	CHECK(String("-999999999999999999999999999999999999").to_uint64() == 0);
+	ERR_PRINT_ON;
+
+	// Test Clamp
+	CHECK(String("-123").to_uint64(true) == 0);
+	CHECK(String("999999999999999999999999999999999999").to_uint64(true) == UINT64_MAX);
+	CHECK(String("-123").to_uint64(true) == 0);
+	CHECK(String("- 22").to_uint64(true) == 0);
+	CHECK(String("-0").to_uint64(true) == 0);
+	CHECK(String("--45").to_uint64(true) == 0);
+	CHECK(String("12a34").to_uint64(true) == 12);
+	CHECK(String("abc").to_uint64(true) == 0);
+
+	// Test different string types
+	CHECK(String::to_uint64("12345") == 12345);
+	CHECK(String::to_uint64(L"12345") == 12345);
+	CHECK(String::to_uint64(U"12345") == 12345);
+
+	// Test length parameter
+	CHECK(String::to_uint64("12345", 3) == 123);
+	CHECK(String::to_uint64("12.345", 2) == 12);
+	CHECK(String::to_uint64("12345", 10) == 12345);
+}
```

Comment:
I'm missing test cases for incorrect strings, like "+ ", as I listed some above that currently fail :)

Reply:
I can add those